File: hw/pastor/u1/hw08FileManagement/chickenInJavaScript/ec.espe.edu.chickenFarmSimulator.model/ChickenCoop.js
const Chicken = require("./Chicken");

class ChickenCoop{
    constructor(chickenCoopNumber){
        this._chickenCoopNumber = chickenCoopNumber;
        this._chickenCoops = [];    
    }

    addChicken(newChicken){
        if (newChicken instanceof Chicken){
            this._chickenCoops.push(newChicken);
        } 
    }

    showChickens(){
        console.log("---------------------------------------------------------------");
        console.log(`Free-range chickens in coop number  ${this._chickenCoopNumber}:`);
        console.log("---------------------------------------------------------------");
        if (!this._chickenCoops.length){
            console.log("The chicken coop is empty");
            return;
        }

        for (const chicken of this._chickenCoops){
            console.log(chicken.toString());
        }
    }

    getChickenCoopNumber(){
        return this._chickenCoopNumber;
    }

    setChickenCoopNumber(chickenCoopNumber){
        this._chickenCoopNumber = chickenCoopNumber;
    }

    getChickenCoop(){
        return this._chickenCoops;
    }

    setChickenCoop(chickenCoops){
        this._chickenCoops = chickenCoops;
    }
}

module.exports = ChickenCoop;

File: hw/pastor/u1/hw08FileManagement/chickenInPython/ChickenCoop.py
from Chiken import Chicken
class ChickenCoop:
    def __init__(self, chicken_coop_number):
        self.chicken_coop_number = chicken_coop_number
        self.chicken_coops = []

    def chicken_coop_number(self):
        return self._chicken_coop_number
        
    def chicken_coops(self):
        return self.chicken_coops
    
    def chicken_coops(self, chicken_coops):
        self._chicken_coops = chicken_coops

    def chicken_coop_number(self, chicken_coop_number):
       self._chicken_coop_number = chicken_coop_number

    def add_chicken(self, new_chicken):
        if isinstance(new_chicken, Chicken):
            self.chicken_coops.append(new_chicken)

    def show_chickens(self):
        print("---------------------------------------------------------------")
        print(f"Free-range chickens in coop number {self.chicken_coop_number}:")
        print("---------------------------------------------------------------")
        if not self.chicken_coops:
            print("The chicken coop is empty")
            return
        for i, new_chicken in enumerate(self.chicken_coops):
            print(f"{new_chicken}")



    
